src/api/utils/tradeQuote/tradequote.ts

```diff
@@ -0,0 +1,404 @@
+  }
+
+  private tokenDisplayAmount(amount: BigNumber, address: Address): string {
+    return this.normalizeTokenAmount(amount, address).toString();
+  }
+
+  private tokenResponse(address: Address): TokenResponse {
+    const tokenEntry = this.tokenMap[address];
+    return {
+      symbol: tokenEntry.symbol,
+      name: tokenEntry.name,
+      address,
+      decimals: tokenEntry.decimals,
+    };
+  }
+
+  private chainCurrencyAddress(chainId: number): Address {
+    switch (chainId) {
+      case 1:   return '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'; // WETH
+      case 137: return '0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270'; // WMATIC
+      default: throw new Error(`chainId: ${chainId} is not supported`);
+    }
+  }
+
+  private normalizeTokenAmount(amount: BigNumber, address: Address): number {
+    const tokenScale = BigNumber.from(10).pow(this.tokenMap[address].decimals);
+    return FixedNumber.from(amount).divUnsafe(FixedNumber.from(tokenScale)).toUnsafeFloat();
+  }
+
+  private tokenPriceUsd(amount: BigNumber, address: Address, coinPrices: CoinGeckoCoinPrices): string {
+    const coinPrice = coinPrices[address][USD_CURRENCY_CODE];
+    const normalizedAmount = this.normalizeTokenAmount(amount, address) * coinPrice;
+    return new Intl.NumberFormat('en-US', {style: 'currency', currency: 'USD'}).format(normalizedAmount);
+  }
+
+  private formatAsPercentage(percentage: number): string {
+    return percentage.toFixed(2) + '%';
+  }
+
+  private totalGasCost(gasPrice: number, gas: number): number {
+    return (gasPrice / 1e9) * gas;
+  }
+
+  private gasCostsUsd(
+    gasPrice: number,
+    gas: number,
+    coinPrices: CoinGeckoCoinPrices,
+    chainId: number
+  ): string {
+    const totalGasCost = this.totalGasCost(gasPrice, gas);
+    const chainCurrencyAddress = this.chainCurrencyAddress(chainId);
+    const coinPrice = coinPrices[chainCurrencyAddress][USD_CURRENCY_CODE];
+    const cost = totalGasCost * coinPrice;
+
+    // Polygon prices are low - using 4 significant digits here so something besides zero appears
+    const options = {
+      style: 'currency',
+      currency: 'USD',
+      maximumSignificantDigits: (chainId === 137) ? 4 : undefined,
+    };
+    return new Intl.NumberFormat('en-US', options).format(cost);
+  }
+
+  private gasCostsChainCurrency(gasPrice: number, gas: number, chainId: number): string {
+    const chainCurrency = this.chainCurrency(chainId);
+    const totalGasCostText = this.totalGasCost(gasPrice, gas).toFixed(7).toString();
+    return `${totalGasCostText} ${chainCurrency}`;
+  }
+
+  private chainCurrency(chainId: number): string {
+    switch (chainId) {
+      case 1:   return 'ETH';
+      case 137: return 'MATIC';
+      default:  return '';
+    }
+  }
+
+  private estimateGasCost(zeroExGas: number): number {
+    const gas = zeroExGas + this.largeTradeGasCostBase;
+    const gasCostBuffer = (100 + this.tradeQuoteGasBuffer) / 100;
+    return Math.floor(gas * gasCostBuffer);
```

Comment:
Need to estimate gas using the trade module w/ from `{ from: manager }`

https://github.com/SetProtocol/set-node/blob/master/src/controllers/v2/gasCostsController.ts